index.js

```diff
@@ -1,20 +1,55 @@
 'use strict';
-const crypto = require('crypto');
+const {Worker} = require('worker_threads');
 
-const create = algorithm => async (buffer, options) => {
+let worker; // Lazy
+let taskIdCounter = 0;
+const tasks = new Map();
+
+const taskWorker = (value, transferList) => new Promise(resolve => {
+	const id = taskIdCounter++;
+	tasks.set(id, resolve);
+	if (worker === undefined) {
+		worker = new Worker('./thread.js');
+		worker.on('message', msg => {
+			const task = tasks.get(msg.id);
+			tasks.delete(msg.id);
+			if (tasks.size === 0) {
+				worker.unref();
+			}
+
+			task(msg.value);
+		});
+		worker.on('error', err => {
+			// Any error here is effectively an equivalent of segfault, and have no scope, so we just throw it on callback level
+			throw err;
+		});
+	}
+
+	worker.postMessage({id, value}, transferList);
+});
+
+const create = algorithm => async (src, options) => {
 	options = {
 		outputFormat: 'hex',
 		...options
 	};
 
-	const hash = crypto.createHash(algorithm);
-	hash.update(buffer, typeof buffer === 'string' ? 'utf8' : undefined);
+	let buffer;
+	if (typeof src === 'string') {
+		// Saving one copy operation by writing string to buffer right away and then transfering buffer
+		buffer = new ArrayBuffer(Buffer.byteLength(src, 'utf8'));
+		Buffer.from(buffer).write(src, 'utf8');
```

Comment:
Why not just:

```suggestion
		buffer = Buffer.from(src);
```

Reply:
`Buffer.from(string)` may allocate buffer from buffer pool depending on the size. Then we will either have to pass it without transfer(with copy), or transfer and wait for some undefined behavior(I didn't test, but I reckon buffer pool will be destroyed, anyway, not defined, so may change any time) to happen.

From my tests, copy itself takes around 1/10 time hashing takes and happens synchronously, so not the best proposition.

Reply:
Checked nodejs source. `Buffer.from(string)` does use buffer pool, and as far as I can tell, implementation of transfer doesn't check for it.